### PR TITLE
Bxmsdoc 2569.roles.x

### PR DESCRIPTION
**Pull request by gsheldon**

### CODE REVIEW

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc [hunk 1]
```diff
@@ -1,2 +1,46 @@
...
+
+. Provide the following role configuration in the request body:
++
+[source,java]
+----
+{
+  "case-data" : {  },
+  "case-user-assignments" : {
+    "owner" : "maciek",
+    "manager" : "maciek"
```
Comment (mswiderski) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc:32]: I'd use different actors for owner and manager just for better readability

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc [hunk 2]
```diff
@@ -1,2 +1,46 @@
...
+  },
+  "case-group-assignments" : {
+    "supplier" : "IT"
+ }
+}
+----
+
+This starts an empty case with defined roles. Two of the roles are user assignments (`owner` and `manager`) and third is a group assignment (`supplier`).
```
Comment (mswiderski) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc:40]: it's not really empty case as there are auto start activities which are directly started and ready to be worked on

File: docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc [hunk 3]
```diff
@@ -1,8 +1,40 @@
...
+Case roles provide an additional layer of abstraction for user participation in case handling. Roles, users, and groups are used for different purposes in case management: 
 
-Roles are optional, and not a case requirement, however, it is considered to be best practice to use roles within case definitions.
+Role::
+Roles drive the authorization for the case instance, and can be used for user activities assignment. The only role that is predefined is `Owner`. A user, a group, or groups can be assigned to the owner role. The owner is whoever the case belongs to. Roles are not restricted to a single set of people or groups as part of case definition.  Roles should always be used for task assignment rather than assigning a specific user or group in order to ensure the case remains dynamic.
```
Comment (mswiderski) [inline docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc]: no roles are predefined so no Owner roles is there by default

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-itorders-roles-proc.adoc [hunk 4]
```diff
@@ -1,29 +1,21 @@
...
 
-.Procedure 
-. *Define the case roles*
+Case roles 
+
+The IT Orders sample case management project includes the following roles:
 +
-Click image:3898.png[] to open the *Properties* menu on the right side of the designer and open the *Editor for Case Roles*. Click *Add Case Role* to add case roles. 
+* `owner` - the employee who is making the hardware order request. There can be only one of these roles.
+* `manager` - the employee's manager; the person who will approve or deny the requested hardware. There is only one manager in the IT Orders project.
```
Comment (mswiderski) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-itorders-roles-proc.adoc:9]: same comment here - manager is a role and not really in the project so it must be assigned when starting a case instance

Comment (gsheldon) [reply docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-itorders-roles-proc.adoc:9]: done in 2568 branch.

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc (hunk 1 above)
Comment (gsheldon) [reply docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc:32]: Changed to "cam" as that's the case manager persona.

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc (hunk 2 above)
Comment (gsheldon) [reply docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-role-authorization-proc.adoc:40]: changed to "This starts a new case with defined roles, as well as autostart activities, which are started and ready to be worked on."

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-itorders-roles-proc.adoc [hunk 5]
```diff
@@ -9,21 +9,15 @@ Click image:3898.png[] to open the *Properties* menu on the right side of the de
 .ITOrders Case Roles
 image::case_roles.png[Case Roles]
 +
+<<<<<<< HEAD
 * `owner` - the person requesting the hardware. There can be only one of these roles.
```
Comment (michelehaglund) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-itorders-roles-proc.adoc]: Is the owner role also set to cardinality 1? If so, I'd same the same as you noted for the manager. Just a late night thought.

File: docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc [hunk 6]
```diff
@@ -1,8 +1,39 @@
...
```
Comment (michelehaglund) [inline docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc]: Should it be: Roles drive the authorization for the case instance, and can be used for user activity assignments ?

File: docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc [hunk 7]
```diff
@@ -1,8 +1,39 @@
...
+
+Group::
+A group is a number of people who are able to carry out a particular task or have a set of responsibilities. There can be any number of people assigned to a group, and a group can be assigned to a role. Members of a group can be added or changed at any time, which is why a group should never be hardcoded to a particular role.
```
Comment (michelehaglund) [inline docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc]: should it be:
A group is a number of people who are able to carry out a particular task or have a set of specified responsibilities

File: docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc [hunk 8]
```diff
@@ -1,8 +1,39 @@
...
\ No newline at end of file
+Role assignments can be given when the case starts or while it is active at any time, and can be changed at any point while the case instance is active. Case Management users should always assign roles to task assignments rather than actual user or group names. This allows the case to be as dynamic as possible, as actual user or group assignment can happen as late as possible, and can also be changed.
+
+When the roles are defined, case management must ensure that these are not hardcoded to single set of people or groups as part of case definition and that they can differ for each case instance. This is where case role assignments become important. Role assignments can be assigned in the following ways:
++
+* given when case starts
+* set at any given point in time while case is active
+* removed at any given point in time while case is active
+
```
Comment (michelehaglund) [inline docs/product-user-guide/src/main/asciidoc/case-management-roles-con.adoc]: May be better as "the case" for the above three bullets. No biggie.

**Pull request merged**